### ISSUE TITLE
Host toggling between Disconnected and Reconnecting states after upgrade

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#7849]
**Rancher Versions:**
Server:v1.4.1 to v1.5.0-alpha1

**Docker Version:**1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**Ubuntu, cloud

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**Single node

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**Cattle

**Steps to Reproduce:**
1. Create a 3 host setup in v1.4.1
2. Create few services, lbs
3. Upgrade to v1.5.0-alpha1

After upgrade, one of the hosts toggles between Disconnected and Reconnecting states.

**Comment by StrongMonkey** [comment]
It appears that the host is nearly running out of memory and rancher-metadata just takes 45% of memory on the host. Everything is slow in that host. @soumyalj Can you try another upgrade to make sure it is not happening randomly?

**Comment by StrongMonkey** [comment]
![screen shot 2017-02-16 at 12 39 21 pm](https://cloud.githubusercontent.com/assets/9203378/23043210/d4de998c-f458-11e6-9839-7f9f9e457607.png)

**Comment by StrongMonkey** [comment]
```time="2017-02-16T21:46:24Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-16T21:46:25Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:46:25Z" level=info msg="Downloaded in 4.518728ms"
time="2017-02-16T21:46:25Z" level=error msg="Failed to download and reload metadata: non-200 response 401: {\"id\":\"8602b146-e480-4d9e-b35c-14b720e7e1c5\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:25Z" level=error msg="Error subscribing to events: websocket: bad handshake" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:46:25Z" level=error msg="Got error response" responseHeaders=map[Expires:[Thu, 01 Jan 1970 00:00:00 GMT] X-Rancher-Version:[v1.5.0-alpha1] X-Api-Schemas:[http://130.211.204.68:8080/v1/schemas] Content-Type:[application/json; charset=utf-8] Server:[Jetty(9.2.11.v20150529)] Date:[Thu, 16 Feb 2017 21:46:25 GMT] Set-Cookie:[PL=rancher;Path=/]] status="401 Unauthorized" statusCode=401
time="2017-02-16T21:46:25Z" level=error msg="Error response: {\"id\":\"ff4abf27-d58d-412f-9a0c-3c902769e535\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:25Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-16T21:46:26Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:46:26Z" level=error msg="Error subscribing to events: websocket: bad handshake" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:46:26Z" level=error msg="Got error response" responseHeaders=map[Date:[Thu, 16 Feb 2017 21:46:26 GMT] Set-Cookie:[PL=rancher;Path=/] Expires:[Thu, 01 Jan 1970 00:00:00 GMT] X-Rancher-Version:[v1.5.0-alpha1] X-Api-Schemas:[http://130.211.204.68:8080/v1/schemas] Content-Type:[application/json; charset=utf-8] Server:[Jetty(9.2.11.v20150529)]] status="401 Unauthorized" statusCode=401
time="2017-02-16T21:46:26Z" level=error msg="Error response: {\"id\":\"19c632b6-ce4e-4555-b3b0-8301f1d3fb2d\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:26Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-16T21:46:26Z" level=info msg="Downloaded in 8.981872ms"
time="2017-02-16T21:46:26Z" level=error msg="Failed to download and reload metadata: non-200 response 401: {\"id\":\"90b4d9ff-bd5f-4f6f-a174-a57a4b9468d8\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:27Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:46:28Z" level=error msg="Error subscribing to events: websocket: bad handshake" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:46:28Z" level=error msg="Got error response" responseHeaders=map[Server:[Jetty(9.2.11.v20150529)] Date:[Thu, 16 Feb 2017 21:46:28 GMT] Set-Cookie:[PL=rancher;Path=/] Expires:[Thu, 01 Jan 1970 00:00:00 GMT] X-Rancher-Version:[v1.5.0-alpha1] X-Api-Schemas:[http://130.211.204.68:8080/v1/schemas] Content-Type:[application/json; charset=utf-8]] status="401 Unauthorized" statusCode=401
time="2017-02-16T21:46:28Z" level=error msg="Error response: {\"id\":\"458d2d63-76d2-4454-bd71-d42367aa0720\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:28Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-16T21:46:28Z" level=info msg="Downloaded in 923.999795ms"
time="2017-02-16T21:46:28Z" level=error msg="Failed to download and reload metadata: non-200 response 401: {\"id\":\"1a4018fa-ed32-4cef-afbf-868e37e04f9a\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:29Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:46:29Z" level=error msg="Error subscribing to events: websocket: bad handshake" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:46:29Z" level=error msg="Got error response" responseHeaders=map[Date:[Thu, 16 Feb 2017 21:46:29 GMT] Set-Cookie:[PL=rancher;Path=/] Expires:[Thu, 01 Jan 1970 00:00:00 GMT] X-Rancher-Version:[v1.5.0-alpha1] X-Api-Schemas:[http://130.211.204.68:8080/v1/schemas] Content-Type:[application/json; charset=utf-8] Server:[Jetty(9.2.11.v20150529)]] status="401 Unauthorized" statusCode=401
time="2017-02-16T21:46:29Z" level=error msg="Error response: {\"id\":\"12ffd734-c370-42dc-941c-d74567accc50\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:29Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-16T21:46:29Z" level=info msg="Downloaded in 4.455944ms"
time="2017-02-16T21:46:29Z" level=error msg="Failed to download and reload metadata: non-200 response 401: {\"id\":\"32246f99-3f80-4e21-9406-2b96d9691d20\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:30Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:46:30Z" level=info msg="Downloaded in 4.509868ms"
time="2017-02-16T21:46:30Z" level=error msg="Failed to download and reload metadata: non-200 response 401: {\"id\":\"5b2e46e2-3fb2-4af7-af8b-39af0f445b2c\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:30Z" level=error msg="Error subscribing to events: websocket: bad handshake" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:46:30Z" level=error msg="Got error response" responseHeaders=map[Content-Type:[application/json; charset=utf-8] Server:[Jetty(9.2.11.v20150529)] Date:[Thu, 16 Feb 2017 21:46:30 GMT] Set-Cookie:[PL=rancher;Path=/] Expires:[Thu, 01 Jan 1970 00:00:00 GMT] X-Rancher-Version:[v1.5.0-alpha1] X-Api-Schemas:[http://130.211.204.68:8080/v1/schemas]] status="401 Unauthorized" statusCode=401
time="2017-02-16T21:46:30Z" level=error msg="Error response: {\"id\":\"51bde603-d361-4443-9965-a83b3ebd8661\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:30Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-16T21:46:31Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:46:31Z" level=info msg="Downloaded in 191.368323ms"
time="2017-02-16T21:46:31Z" level=error msg="Failed to download and reload metadata: non-200 response 401: {\"id\":\"6253541a-2cbf-4077-86db-0b9c59e867fa\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:33Z" level=error msg="Error subscribing to events: websocket: bad handshake" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=config.update&eventNames=ping"
time="2017-02-16T21:46:33Z" level=error msg="Got error response" responseHeaders=map[Server:[Jetty(9.2.11.v20150529)] Date:[Thu, 16 Feb 2017 21:46:31 GMT] Set-Cookie:[PL=rancher;Path=/] Expires:[Thu, 01 Jan 1970 00:00:00 GMT] X-Rancher-Version:[v1.5.0-alpha1] X-Api-Schemas:[http://130.211.204.68:8080/v1/schemas] Content-Type:[application/json; charset=utf-8]] status="401 Unauthorized" statusCode=401
time="2017-02-16T21:46:33Z" level=error msg="Error response: {\"id\":\"a51a9a9d-1608-456f-a3db-b27597b0ca0d\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:46:33Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-16T21:46:34Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:46:39Z" level=info msg="Downloaded in 4.578278694s"
time="2017-02-16T21:47:00Z" level=error msg="Failed to download and reload metadata: non-200 response 401: {\"id\":\"c9bfa38e-0c48-499b-80ad-646edf218195\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":401,\"code\":\"Unauthorized\",\"message\":\"Unauthorized\",\"detail\":null,\"baseType\":\"error\"}"
time="2017-02-16T21:47:23Z" level=error msg="Error subscribing to events: unexpected EOF" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:47:40Z" level=error msg="Exiting subscriber: unexpected EOF"
time="2017-02-16T21:49:06Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:55:06Z" level=error msg="Failed to download and reload metadata: Get http://130.211.204.68:8080/v1/configcontent/metadata-answers: dial tcp 130.211.204.68:8080: i/o timeout"
time="2017-02-16T21:56:17Z" level=error msg="Error subscribing to events: unexpected EOF" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:56:33Z" level=error msg="Exiting subscriber: unexpected EOF"
time="2017-02-16T21:56:59Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-16T21:57:57Z" level=error msg="Failed to download and reload metadata: Get http://130.211.204.68:8080/v1/configcontent/metadata-answers: EOF"
time="2017-02-16T21:58:10Z" level=error msg="Error subscribing to events: unexpected EOF" subscribeUrl="ws://130.211.204.68:8080/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-16T21:58:11Z" level=error msg="Exiting subscriber: unexpected EOF"
time="2017-02-16T21:58:12Z" level=info msg="Initializing event router" workerCount=3```

**Comment by alena1108** [comment]
@StrongMonkey the sequence of events was as follows:

* metadata service went to upgrading state. The old instance got removed on cattle side. Request to create a new instance was issued.
* meanwhile host went to disconnected state by some reason, and it stayed in this state.
* old metadata instance on the host could no longer connect to wsp/rancher as it no longer has agent credentials assigned to it. It gets 401 error and retries the next second.
* as host stays in disconnected state, metadata instance just keeps on trying to establish wsp session.

Potential bug on event-subscriber/metadata side: on any error - including 401 - websocket proxy connection wasn't closed. That bug was fixed with PR referencing this bug.

@StrongMonkey to recover from the issue, just remove the problematic metadata container.

**Comment by junkiebev** [comment]
I have the same issue 401 issue after an aborted upgrade and it is affecting all of my environments and docker hosts even after a DB roll back. Which container(s) am I meant to remove?

**Comment by alena1108** [comment]
@junkiebev containers using rancher/metadata image should be removed as they can potentially expose memory leaks. We are addressing this bug in metadata template for 1.4.1 and 1.5 Rancher releases

**Comment by cjellick** [comment]
@alena1108 I'm going to unassign @StrongMonkey from this. If you want to reassign him, go ahead. I'm just trying to get a handle on his 1.5 issues.

**Comment by alena1108** [comment]
@sangeethah might be a bit tricky to test it, but we can try to test it as a part of Rancher upgrade. So you upgrade rancher, but hold on upgrading metadata service. See if getting a lot of 401 errors leads to the high memory consumption by metadata container.

**Comment by mattdodge** [comment]
I too am seeing the rancher/metadata container eating up a large portion of memory inside of the network-services stack (version 0.0.10). @alena1108 when you say we shouldn't be using the rancher/metadata image does that apply to network services too? It appears I'm running the most up to date version of that stack (Rancher 1.4.1), but I think I'm seeing the memory leaks there. It also spits out error logs at an alarming rate, typically looking something like this:
```
time="2017-02-03T14:51:48Z" level=error msg="Exiting subscriber: websocket: bad handshake"
time="2017-02-03T14:51:49Z" level=info msg="Initializing event router" workerCount=3
time="2017-02-03T14:51:49Z" level=info msg="Downloaded in 3.953593ms"
time="2017-02-03T14:51:49Z" level=error msg="Failed to download and reload metadata: non-200 response 502: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>awselb/2.0</center>\r\n</body>\r\n</html>\r\n"
time="2017-02-03T14:51:49Z" level=error msg="Error subscribing to events: websocket: bad handshake" subscribeUrl="wss://myrancherhost/v1/subscribe?eventNames=ping&eventNames=config.update"
time="2017-02-03T14:51:49Z" level=error msg="Got error response" responseHeaders=map[Server:[awselb/2.0] Date:[Fri, 03 Feb 2017 14:51:49 GMT] Content-Type:[text/html] Content-Length:[171] Connection:[keep-alive] Set-Cookie:[AWSALB=NeJd7+PszWpzu78J3v0Sf6G00ZQabzFGpZ1l5g9DuMmgmpN1unl1hORh8WGv6pC/b4/Q3KqYrwpKOPIZhfo3HA634lS50BkCQmnv/aAz6+IgrGe+i/1KGnlAmGQH; Expires=Fri, 10 Feb 2017 14:51:49 GMT; Path=/]] status="502 Bad Gateway" statusCode=502
time="2017-02-03T14:51:49Z" level=error msg="Error response: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>awselb/2.0</center>\r\n</body>\r\n</html>\r\n"
```

I did recently switch from an ALB to an ELB but the load balancer seems healthy, not sure exactly where the 502s are coming from.

**Comment by santo74** [comment]
I too am getting out of memory issues after upgrading from Rancher 1.1 to Rancher 1.4 (actually a new install rather than an upgrade of the existing server).
After several hours the rancher-metadata container is using up to 40% of memory, which causes  swapping because I'm also running some java processes on that system that take up some memory.
After some time, the system completely hangs and isn't reachable anymore or it just crashes with an out of memory error.

Stopping the rancher-metadata container on the server brings the server back to a normal state (via the gui doesn't have any effect as the server is not very responsive then) 

My servers should go in production really soon (next week), so it's a critical issue for me.
Hopefully it can get fixed in time for v1.4.2

**Comment by santo74** [comment]
This is what's happening in the log of the metadata-service:

```
time="2017-02-22T02:25:14Z" level=info msg="Downloaded in 344.27687ms" 
time="2017-02-22T02:25:14Z" level=info msg="Loading answers" 
time="2017-02-22T02:25:16Z" level=info msg="Loaded answers" 
time="2017-02-22T02:25:16Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1396-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T02:25:16Z" level=info msg="Downloaded in 205.295478ms" 
time="2017-02-22T02:25:16Z" level=info msg="Loading answers" 
time="2017-02-22T02:25:17Z" level=info msg="Loaded answers" 
time="2017-02-22T02:25:17Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1396-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T06:42:01Z" level=info msg="Hit websocket pong timeout. Last websocket ping received at 2017-02-22 06:40:47.473102786 +0000 UTC. Closing connection." 
time="2017-02-22T06:47:00Z" level=warning msg="websocket closed: write tcp 172.17.0.2:47576->104.27.187.85:443: i/o timeout" 
time="2017-02-22T06:53:15Z" level=info msg="Hit websocket pong timeout. Last websocket ping received at 2017-02-22 06:40:47.473102786 +0000 UTC. Closing connection." 
time="2017-02-22T06:56:04Z" level=info msg="Hit websocket pong timeout. Last websocket ping received at 2017-02-22 06:40:47.473102786 +0000 UTC. Closing connection." 
time="2017-02-22T06:59:08Z" level=info msg="Initializing event router" workerCount=3 
time="2017-02-22T07:16:15Z" level=error msg="Failed to download and reload metadata: Get https://rancher.example.com/v1/configcontent/metadata-answers: dial tcp: i/o timeout" 
time="2017-02-22T07:43:18Z" level=error msg="Error subscribing to events: EOF" subscribeUrl="wss://rancher.example.com/v1/subscribe?eventNames=ping&eventNames=config.update" 
time="2017-02-22T07:45:54Z" level=error msg="Exiting subscriber: EOF" 
time="2017-02-22T07:47:06Z" level=info msg="Initializing event router" workerCount=3 
time="2017-02-22T07:55:40Z" level=error msg="Failed to download and reload metadata: Get https://rancher.example.com/v1/configcontent/metadata-answers: dial tcp: i/o timeout" 
time="2017-02-22T08:41:24Z" level=error msg="Error subscribing to events: dial tcp: lookup rancher.example.com on 178.22.71.56:53: dial udp 178.22.71.56:53: i/o timeout" subscribeUrl="wss://rancher.example.com/v1/subscribe?eventNames=ping&eventNames=config.update" 
time="2017-02-22T08:41:24Z" level=error msg="Exiting subscriber: dial tcp: lookup rancher.example.com on 178.22.71.56:53: dial udp 178.22.71.56:53: i/o timeout" 
time="2017-02-22T08:41:25Z" level=info msg="Initializing event router" workerCount=3 
time="2017-02-22T08:41:26Z" level=info msg="Connection established" 
time="2017-02-22T08:41:26Z" level=info msg="Starting websocket pings" 
time="2017-02-22T08:41:27Z" level=info msg="Downloaded in 2.073736286s" 
time="2017-02-22T08:41:27Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:27Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:27Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1410-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:31Z" level=info msg="Downloaded in 901.37011ms" 
time="2017-02-22T08:41:31Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:31Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:31Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1417-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:31Z" level=info msg="Downloaded in 484.834963ms" 
time="2017-02-22T08:41:31Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:32Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:32Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1417-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:33Z" level=info msg="Downloaded in 1.081827427s" 
time="2017-02-22T08:41:33Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:33Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:33Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1418-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:33Z" level=info msg="Downloaded in 401.407288ms" 
time="2017-02-22T08:41:33Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:34Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:34Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1418-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:35Z" level=info msg="Downloaded in 1.425960764s" 
time="2017-02-22T08:41:36Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:36Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:36Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1419-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:37Z" level=info msg="Downloaded in 908.334058ms" 
time="2017-02-22T08:41:37Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:37Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:37Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1424-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:37Z" level=info msg="Downloaded in 209.744193ms" 
time="2017-02-22T08:41:37Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:37Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:37Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1424-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:40Z" level=info msg="Downloaded in 409.304832ms" 
time="2017-02-22T08:41:40Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:40Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:40Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1425-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:40Z" level=info msg="Downloaded in 203.720544ms" 
time="2017-02-22T08:41:40Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:40Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:40Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1425-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:42Z" level=info msg="Downloaded in 598.683963ms" 
time="2017-02-22T08:41:42Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:42Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:42Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1426-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:43Z" level=info msg="Downloaded in 910.043285ms" 
time="2017-02-22T08:41:43Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:43Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:43Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1426-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:44Z" level=info msg="Downloaded in 941.424923ms" 
time="2017-02-22T08:41:44Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:44Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:44Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1428-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:46Z" level=info msg="Downloaded in 1.121258752s" 
time="2017-02-22T08:41:46Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:46Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:46Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1431-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:47Z" level=info msg="Downloaded in 1.709989457s" 
time="2017-02-22T08:41:47Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:48Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:48Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1433-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:48Z" level=info msg="Downloaded in 517.193304ms" 
time="2017-02-22T08:41:48Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:48Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:48Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1434-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:48Z" level=info msg="Downloaded in 183.017923ms" 
time="2017-02-22T08:41:48Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:48Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:48Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1434-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:41:49Z" level=info msg="Downloaded in 81.687638ms" 
time="2017-02-22T08:41:49Z" level=info msg="Loading answers" 
time="2017-02-22T08:41:49Z" level=info msg="Loaded answers" 
time="2017-02-22T08:41:49Z" level=info msg="Applied https://rancher.example.com/v1/configcontent/metadata-answers?version=1434-5cc6db6887beb3525be1bd696730805e" 
time="2017-02-22T08:42:47Z" level=info msg="Hit websocket pong timeout. Last websocket ping received at 2017-02-22 08:42:06.243112439 +0000 UTC. Closing connection." 
time="2017-02-22T08:42:49Z" level=warning msg="websocket closed: websocket: write timeout" 
time="2017-02-22T08:43:20Z" level=info msg="Hit websocket pong timeout. Last websocket ping received at 2017-02-22 08:42:06.243112439 +0000 UTC. Closing connection." 
time="2017-02-22T08:43:25Z" level=info msg="Hit websocket pong timeout. Last websocket ping received at 2017-02-22 08:42:06.243112439 +0000 UTC. Closing connection." 
```

So for some reason something goes wrong at 06:42:01.
2 hours later, the connection is re-established but now it starts filling the logs and memory usage increases

So 2 things here:
why in the first place are there connection issues?
why is it filling up the logs and memory (after reconnecting)?

**Comment by soumyalj** [comment]
Tested with upgrading rancher from v1.4.1 with rancher/metadata:v0.7.2 (which has the metadata fix backported) to v1.5.0-rc1. No infra services were upgraded and left the setup for more than 48 hours. Did not see this issue. The memory usage for rancher-metadata container on the hosts was around 1%.
